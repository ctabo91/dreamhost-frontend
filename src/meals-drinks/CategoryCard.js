import React from "react";
import { Link } from "react-router-dom";
import "./CategoryList.css";

/** A card showing category "name" and a "count" for how many recipes that category contains
 * 
 * Card is a link that redirects to RecipesList with category name passed as the :category param.
 */

function CategoryCard({ category, count, item }) {
    return (
        <div className="col-4 mb-4">
            <Link className="CategoryCard card border-0" to={`/recipes/${item}/${encodeURIComponent(category)}`}>
                <div className="card-body text-center pl-0 pr-0">
                    <h5 className="card-title">{category.toUpperCase()}</h5>
                    <p className="card-text"><small>{count} recipes</small></p>
                </div>
            </Link>
        </div>
    );
}

export default CategoryCard;