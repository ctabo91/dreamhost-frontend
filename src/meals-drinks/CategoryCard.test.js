import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CategoryCard from './CategoryCard';

const mockCategory = {
  category: 'Category 1',
  count: 5,
};

describe('CategoryCard Component', () => {
  it('matches snapshot', () => {
    const{ asFragment } = render(
      <MemoryRouter>
        <CategoryCard category={mockCategory.category} count={mockCategory.count} item="meals" />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders category card correctly', () => {
    render(
      <MemoryRouter>
        <CategoryCard category={mockCategory.category} count={mockCategory.count} item="meals" />
      </MemoryRouter>
    );

    expect(screen.getByText('CATEGORY 1')).toBeInTheDocument();
    expect(screen.getByText('5 recipes')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/recipes/meals/Category%201');
  });

  it('handles special characters in category name', () => {
    render(
      <MemoryRouter>
        <CategoryCard category="Special Category" count={3} item="drinks" />
      </MemoryRouter>
    );

    expect(screen.getByText('SPECIAL CATEGORY')).toBeInTheDocument();
    expect(screen.getByText('3 recipes')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/recipes/drinks/Special%20Category');
  });
});
