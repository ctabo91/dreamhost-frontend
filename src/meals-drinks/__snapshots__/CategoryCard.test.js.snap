// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-4 mb-4"
  >
    <a
      class="CategoryCard card border-0"
      href="/recipes/meals/Category%201"
    >
      <div
        class="card-body text-center pl-0 pr-0"
      >
        <h5
          class="card-title"
        >
          CATEGORY 1
        </h5>
        <p
          class="card-text"
        >
          <small>
            5 recipes
          </small>
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
